superset-frontend/packages/superset-ui-core/src/number-format/factories/createTimeDurationFormatter.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import NumberFormatter from '../NumberFormatter';
+
+export default function createTimeDurationFormatter(
+  config: {
+    description?: string;
+    id?: string;
+    label?: string;
+    multiplier?: number;
+  } = {},
+) {
+  const { description, id, label } = config;
+
+  return new NumberFormatter({
+    description,
+    formatFunc: value => {
+      const v = Math.abs(Math.round(value));
+      const h = Math.floor(v / 60);
+      const m = v % 60;
+      return `${h}:${String(m).padStart(2, '0')}`;
+    },
+    id: id ?? 'duration_format',
+    label: label ?? `Duration formatter`,
+  });
+}
```

Comment:
We already have a dependency on [pretty-ms](https://www.npmjs.com/package/pretty-ms) that achieves this and (see `colonNotation` option).